.github/workflows/publish.yml

```diff
@@ -1,79 +1,79 @@
-      LBHPACKAGESTOKEN: ${{secrets.LBHPACKAGESTOKEN }}
-    outputs:
-      version: ${{ needs.calculate-version.outputs.version }}
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: Build
-        run: docker-compose build hackney-shared-patches-and-areas-test
-      - name: Run tests
-        run: docker-compose run hackney-shared-patches-and-areas-test
-
-  publish-package:
-    name: Publish Package
-    runs-on: ubuntu-latest
-    needs: build-and-test
-    env:
-      VERSION: ${{ needs.build-and-test.outputs.version }}
-      LBHPACKAGESTOKEN: ${{secrets.LBHPACKAGESTOKEN }}
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: Build the Package
-        run: dotnet pack Hackney.Shared.PatchesAndAreas -p:PackageVersion=$VERSION --configuration Release
-      - name: Publish the Package
-        run: |
-          cd Hackney.Shared.PatchesAndAreas/bin/Release
+name: publish-nuget-package
+on:
+  push:
+    branches:
+    - main
+    - feature/**
+    
+jobs:
+  calculate-version:
+    name: Calculate Version
+    runs-on: ubuntu-latest
+    env:
+      LBHPACKAGESTOKEN: ${{secrets.LBHPACKAGESTOKEN }}
+    outputs:
+      version: ${{ steps.gitversion.outputs.nuGetVersionV2 }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Install GitVersion
+        uses: gittools/actions/gitversion/setup@v0.9.7
+        with:
+          versionSpec: '5.x'
+      - name: Determine package version
+        id: gitversion
+        uses: gittools/actions/gitversion/execute@v0.9.7
+        with:
+          useConfigFile: true
+      - name: Display package version
+        run: |
+          echo "Version: $GITVERSION_NUGETVERSIONV2"
+
+  check-code-formatting:
+    name: Check code formatting
+    runs-on: ubuntu-latest
+    needs: calculate-version
+    env:
+      LBHPACKAGESTOKEN: ${{secrets.LBHPACKAGESTOKEN }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install dotnet format
+        run: dotnet tool install dotnet-format --tool-path ./dotnet-format-local/
+      - name: Run formatter check
+        run: ./dotnet-format-local/dotnet-format --check
+
+  build-and-test:
+    name: Build & Test
+    runs-on: ubuntu-latest
+    needs: calculate-version
+    env:
+      LBHPACKAGESTOKEN: ${{secrets.LBHPACKAGESTOKEN }}
+    outputs:
+      version: ${{ needs.calculate-version.outputs.version }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Build
+        run: docker-compose build hackney-shared-patches-and-areas-test
+      - name: Run tests
+        run: docker-compose run hackney-shared-patches-and-areas-test
+
+  publish-package:
+    name: Publish Package
+    runs-on: ubuntu-latest
+    needs: build-and-test
+    env:
+      VERSION: ${{ needs.build-and-test.outputs.version }}
+      LBHPACKAGESTOKEN: ${{secrets.NUGET_KEY }}

```

Comment:
Has this secret changed?

Reply:
This should remain as LBHPACKAGESTOKEN

Reply:
LBHPACKAGESTOKEN has read-only permission so I had to change it to Nuget_key. 
I think the Nuget_key in the lbhackney-it github account has expired which is why, the pipeline was failing. I can't log in to the  lbhackney-it github account as I'm not part of the keeper development team vault so I used one of my own token and will update it again once I get the relevant access.